core/circuit.go

```diff
@@ -0,0 +1,244 @@
+package core
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/util"
+)
+
+var (
+	circuitId int // counter for circuit id
+)
+
+// struct to setup the circuit hierarchy
+type CircuitConfig struct {
+	// Title      string           `mapstructure:"title"`      // printable name
+	Name       string           `mapstructure:"name"`       // unique name, used as reference in lp
+	MaxCurrent float64          `mapstructure:"maxCurrent"` // the max allowed current of this circuit
+	MeterRef   string           `mapstructure:"meter"`      // Charge meter reference
+	Circuits   []*CircuitConfig `mapstructure:"circuits"`   // sub circuits as config reference
+
+	CircuitRef *Circuit // reference to instance
+	vMeter     *VMeter  // virtual meter for the circuit, if needed
+}
+
+// the circuit instances to control the load
+type Circuit struct {
+	log    *util.Logger
+	uiChan chan<- util.Param
+
+	Title         string            // pretty logging
```

Comment:
Title raus. Über injizierten Logger identifizieren. In setup.go:


	for id, cfg := range conf.Circuits {
		log := util.NewLogger("circuit-" + strconv.Itoa(id+1))

		circuitMapNew, vmeterMapNew, err := core.NewCircuitFromConfig(log, cp, cfg)

Reply:
Alle Referenzen auf circuit.Title entfallen damit. Auch sonst können (aktuell) Devices ihren Title nicht loggen.